### ISSUE TITLE
Route Blinding: allow path creator to specify incoming channel

### ISSUE THREAD

**Issue by ellemouton** [lightningnetwork/lnd#8993]
When creating a blinded payment path for an invoice, we should allow the caller to 
select the channel they want to receive on. This will allow them to force a sender to
pay over a specific channel and so gives them an extra liquidity control leaver.

**Comment by alexbosworth** [comment]
Would this be incoming last hop peer or incoming channel?

**Comment by ellemouton** [comment]
> Would this be incoming last hop peer or incoming channel?

good point. So on an instruction level, we tell the peer the next channel to send on. But due to non-strict forwarding, nothing stops them from choosing a diff channel that the one instructed I guess.

**Comment by MPins** [comment]
Interesting! Maybe it is a good chalange to me!

**Comment by ellemouton** [comment]
Feel free to give it a go @MPins :)

**Comment by MPins** [comment]
Thanks! Starting from the [blinded paths proposal](https://github.com/lightning/bolts/pull/765), then the code ;-)

**Comment by ellemouton** [comment]
Might make more sense to make this more general: Allow user to specify the whole path or parts of the path 
(cc @alexbosworth )

**Comment by saubyk** [comment]
Hi @MPins assigned the issue to you

**Comment by MPins** [comment]
Hello all, to do some tests before coding I've just created a network with some 0.18.2 nodes on Polar Lightning and it seems that those nodes do not know blinded paths, but it is available since 0.18.0!!!

**Comment by saubyk** [comment]
Hi @MPins I don't think you'll be able to test blinded paths with 18.2, you'd need 18.3

Release 18 had all the code changes to enable blinded paths, but the implementation and inclusion with the rpcs on LND has been done with 18.3

**Comment by MPins** [comment]
> specify the whole path or parts of the path

To do that we could work with a json object, where the user choose the introduction node (optional), blinded nodes (at least one) and  channels (optional). Example:
{
          "blinded_path":  {
              "introduction_node":  "node_id",
              "blinded_hops":  [
                  {
                      "blinded_node":  "node_id",
                      "channel":  "channel_id"
                  },
                  {
                      "blinded_node":  "node_id",
                      "channel":  "channel_id"
                  },
                  {
                      "blinded_node":  "node_id",
                      "channel":  ""
                  }
              ]
          }
}

But particularly I think it is too much flexibility. From the node runner point of view I think that choosing the last hop node or/and channel would be enough. What do you think?

**Comment by ellemouton** [comment]
@MPins  - sorry for the late reply. 

I think we could just make it general, ie: we allow user to specify a list of peers for the final stretch of the path. If they choose to only specify 1 peer, then that's fine and LND can choose to build multiple paths with that constraint given the other config values. If the user adds a chain of 2 peers, then LND uses that new constraint etc etc

**Comment by MPins** [comment]
Hello ... no problem :slightly_smiling_face: .

It I understood correctly. It is done already. 

Maybe we could take out the option of choosing a list of nodes, and keep a list of channels only.
